usr/lib/pkcs11/ep11_stdll/new_host.c

```diff
@@ -1701,8 +1701,10 @@ CK_RV SC_EncryptUpdate(STDLL_TokData_t * tokdata, ST_SESSION_HANDLE * sSession,
     if (sess->encr_ctx.init_pending) {
         rc = ep11tok_encrypt_init(tokdata, sess, &sess->encr_ctx.mech,
                                   sess->encr_ctx.key);
-        if (rc != CKR_OK)
+        if (rc != CKR_OK) {
             TRACE_DEVEL("ep11tok_encr_init() failed.\n");
+            goto done;
```

Comment:
I dont know what ep11tok_encrypt_init returns in case of deferred init but note that if its not CKR_BUFFER_TOO_SMALL, the session will be cleaned up at 'done'.

Reply:
ah okay. cleaning up is probably what you want here, because there will be no final.

Reply:
yes that's intended.